File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] m, int k) {
        int i=0;
        int j=m[0].length-1;
        while(i<=m.length-1&&j>=0)
        {
            if(k==m[i][j])
            {
                return true;
            }
            else if(k<m[i][j])
            {
                j--;
            }
            else{
                i++;
            }
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.c


bool searchMatrix(int** m, int ms, int *mc, int k){
    for(int i=0;i<ms;i++)
    {
        for(int j=0;j<mc[0];j++)
        {
               if(m[i][j]==k)
               {
                  return true; 
               }
               
        }
    }
    return false; 
}